<launch>
    <rosparam command="load" file="$(find crazyflie_kinect_localization)/config/default.yaml" />

    <node name="cropbox_filter" pkg="simpcl" type="cropbox_filter" output="screen">
        <remap from="points_in" to="/kinect2/qhd/points"/>
        <remap from="points_out" to="/kinect2/qhd/points/filtered/cropbox"/>
    </node>

    <node name="red_color_filter" pkg="simpcl" type="color_filter" output="screen">
        <remap from="points_in" to="/kinect2/qhd/points/filtered/cropbox"/>
        <remap from="points_out" to="/kinect2/qhd/points/filtered/red"/>
    </node>

    <node name="green_color_filter" pkg="simpcl" type="color_filter" output="screen">
        <remap from="points_in" to="/kinect2/qhd/points/filtered/cropbox"/>
        <remap from="points_out" to="/kinect2/qhd/points/filtered/green"/>
    </node>

    <node name="blue_color_filter" pkg="simpcl" type="color_filter" output="screen">
        <remap from="points_in" to="/kinect2/qhd/points/filtered/cropbox"/>
        <remap from="points_out" to="/kinect2/qhd/points/filtered/blue"/>
    </node>

    <node name="centroid_cf1" pkg="simpcl" type="get_centroid" output="screen">
        <remap from="points_in" to="/kinect2/qhd/points/filtered/red"/>
        <remap from="centroid" to="/cf1/pose"/>
    </node>

    <node name="centroid_cf2" pkg="simpcl" type="get_centroid" output="screen">
        <remap from="points_in" to="/kinect2/qhd/points/filtered/green"/>
        <remap from="centroid" to="/cf2/pose"/>
    </node>

    <node name="centroid_cf3" pkg="simpcl" type="get_centroid" output="screen">
        <remap from="points_in" to="/kinect2/qhd/points/filtered/blue"/>
        <remap from="centroid" to="/cf3/pose"/>
    </node>

</launch>
